nucypher/cli/commands/ursula.py

```diff
@@ -392,10 +392,15 @@ def run(general_config, character_options, config_file, interactive, dry_run, me
                                                     metrics_prefix=metrics_prefix,
                                                     listen_address=metrics_listen_address)
 
-    return URSULA.run(emitter=emitter,
-                      start_reactor=not dry_run,
-                      interactive=interactive,
-                      prometheus_config=prometheus_config)
+    # TODO should we just not call run at all for "dry_run"
+    try:
+        URSULA.run(emitter=emitter,
+                   start_reactor=not dry_run,
+                   interactive=interactive,
+                   prometheus_config=prometheus_config)
+    finally:
+        if dry_run:
```

Comment:
Why only in the case of a dry run?

Reply:
If Ursula is starting from the command line, then we don't want to stop it other than for dry-run.

Reply:
Perhaps I am missing something - but why not?  At the end of this test it seems like we can stop the ursula no mater what the situation is.

Reply:
If I'm understanding you correctly - you are correct, but the code change is not to the test, but rather the ursula command module itself.